### PR TITLE
[HttpKernel] Remove EOL when using error_log() in HttpKernel Logger

### PR DESCRIPTION
**Pull request by cyve**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | maybe
| Tickets       | 
| License       | MIT
| Doc PR        | 

Hello there !

Since PR https://github.com/symfony/symfony/pull/36855, when using `Symfony\Component\HttpKernel\Log\Logger`  with argument `$output = null`. The logger use the `error_log()` method to log messages. But in the log file, messages are separated by an empty line because the default formatter (`Logger::format()`) adds an extra EOL char at the end of the message.

This fix prevents the logger to add an extra EOL char when it uses `error_log()`.

Possible BC : if someone use the logger with a custom formatter that already add a EOL char at the end of the message, there will be a empty line between messages in the log file. In my opinion, this is minor enough not to worry about it ;-)

Thanks for your review :-)

### CODE REVIEW

**Comment by cyve** [comment]
Hi, I added a 2nd commit to apply coding standards on code that was not modified by the PR itself. Should I squash them ?

File: src/Symfony/Component/HttpKernel/Tests/Log/LoggerTest.php [hunk 1]
```diff
@@ -196,6 +196,26 @@ public function testFormatter()
...
+        $logger->error('test');
+        $logger->critical('test');
+
+        $expected = [
+            '[error] test',
+            '[critical] test',
+        ];
+
+        foreach ($this->getLogs() as $k => $line) {
+            self::assertThat(1 === preg_match('/\[[\w\/\-: ]+\] '.preg_quote($expected[$k]).'/', $line), self::isTrue(), "\"$line\" do not match expected pattern \"$expected[$k]\"");
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Tests/Log/LoggerTest.php]: $this->assertSame(1, etc)

File: src/Symfony/Component/HttpKernel/Log/Logger.php [hunk 2]
```diff
@@ -44,10 +44,14 @@ public function __construct(string $minLevel = null, $output = null, callable $f
...
-                    case 2: $minLevel = LogLevel::INFO; break;
-                    case 3: $minLevel = LogLevel::DEBUG; break;
+                    case -1: $minLevel = LogLevel::ERROR;
+                        break;
+                    case 1: $minLevel = LogLevel::NOTICE;
+                        break;
+                    case 2: $minLevel = LogLevel::INFO;
+                        break;
+                    case 3: $minLevel = LogLevel::DEBUG;
+                        break;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Log/Logger.php:54]: please revert this change

**Comment by nicolas-grekas** [comment]
Thank you @cyve.
